### PR TITLE
added schemas and base classes for algos

### PR DESCRIPTION
**Pull request by martynov-dm**

### CODE REVIEW

File: dev_packages/chatsky_llm_autoconfig/chatsky_llm_autoconfig/algorithms/topic_graph_generation.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+from langchain.prompts import PromptTemplate
+from langchain_core.output_parsers import JsonOutputParser
+
+from chatsky_llm_autoconfig.graph import BaseGraph, Graph
+
+from langchain.prompts import PromptTemplate
+import os
+
+cycle_graph_generation_prompt = PromptTemplate.from_template(
```
Comment (NotBioWaste905) [inline dev_packages/chatsky_llm_autoconfig/chatsky_llm_autoconfig/algorithms/topic_graph_generation.py:14]: Move into CycleGraphGenerator class

File: dev_packages/chatsky_llm_autoconfig/chatsky_llm_autoconfig/algorithms/topic_graph_generation.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+        generated_graph = chain.invoke(topic)
+
+        return Graph(generated_graph)
+
+    async def ainvoke(self, *args, **kwargs):
+        pass
+
+
+cyc = CycleGraphGenerator()
```
Comment (NotBioWaste905) [inline dev_packages/chatsky_llm_autoconfig/chatsky_llm_autoconfig/algorithms/topic_graph_generation.py]: move instantiation and invocation into if __name__=="__main__"

**Review by NotBioWaste905 — CHANGES_REQUESTED** [review]
Requested minor tweaks

File: dev_packages/chatsky_llm_autoconfig/chatsky_llm_autoconfig/algorithms/topic_graph_generation.py (hunk 2 above)
Comment (martynov-dm) [reply dev_packages/chatsky_llm_autoconfig/chatsky_llm_autoconfig/algorithms/topic_graph_generation.py]: fixed

**Pull request merged**